src/evict/evict_lru.c

```diff
@@ -983,24 +985,34 @@ retry:	while (slot < max_entries && ret == 0) {
 		}
 
 		if (dhandle == NULL)
-			dhandle = SLIST_FIRST(&conn->dhlh);
+			dhandle = STAILQ_FIRST(&conn->dhlh);
 		else {
 			if (incr) {
 				WT_ASSERT(session, dhandle->session_inuse > 0);
 				(void)WT_ATOMIC_SUB4(dhandle->session_inuse, 1);
 				incr = 0;
 			}
-			dhandle = SLIST_NEXT(dhandle, l);
+			dhandle = STAILQ_NEXT(dhandle, l);
 		}
 
 		/* If we reach the end of the list, we're done. */
 		if (dhandle == NULL)
 			break;
 
-		/* Ignore non-file handles, or handles that aren't open. */
-		if (!WT_PREFIX_MATCH(dhandle->name, "file:") ||
+		/*
+		 * Once every 15 runs we do a full walk. If we aren't doing a
+		 * full walk, then we can declare ourselves finished when we
+		 * find the first entry that isn't to be read.
+		 */
+		if (laps % 15 != 0 && dhandle->evict_skip_until > laps) {
+			cache->evict_file_next = NULL;
+			break;
+		}
```

Comment:
What is the point of checking all handles every 15 laps?  Won't that fall out of setting `dhandle->evict_skip_until = laps + 15`?

Or is here because you're not counting the absolute number of laps in the calling code?